src/main/java/com/google/devtools/build/lib/rules/cpp/CcModule.java

```diff
@@ -1538,6 +1548,13 @@ private static void getLegacyArtifactNamePatterns(
     return object != Starlark.NONE ? type.cast(object) : null;
   }
 
+  @Nullable
+  private static <T> ImmutableList<T> nullIfNoneOrEmptySequence(
+      Object object, Class<T> type, String what) throws EvalException {
+    Sequence<T> seq = Sequence.noneableCast(object, type, what);
+    return seq.isEmpty() ? null : seq.getImmutableList();
+  }
+
```

Comment:
This feels weird; returning null instead of an empty list when the input is empty. I would normally expect the opposite (null being transformed into an empty list); is there a reason for this?

Reply:
Yup, in LibraryToLink there is a check for null [](https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/rules/cpp/LibraryToLink.java;l=383?q=librarytolink&ss=bazel%2Fbazel)
but maybe instead of replacing an empty list here I should check inside cc_import.bzl if it's empty and simply pass null there [](https://github.com/bazelbuild/bazel/pull/11696/files#diff-de10b4edef58e36c067b24bf39170475R189). AFAIK when a bazel rule has a label_list attribute then this attribute cannot have a default None value, it must be an empty list [](https://github.com/bazelbuild/bazel/pull/11696/files#diff-de10b4edef58e36c067b24bf39170475R224)